pandas/_libs/tslibs/timestamps.pyi

```diff
@@ -57,7 +55,10 @@ class Timestamp(datetime):
         tzinfo: _tzinfo | None = ...,
         *,
         fold: int | None = ...,
-    ) -> _DatetimeT | NaTType: ...
+    ) -> _DatetimeT: ...
+    # GH 46171
+    # While Timestamp can return pd.NaT, having the constructor return
+    # a Union with NaTType makes things awkward for users of pandas
```

Comment:
I would be inclined to return Any in the public api if there are genuine concerns rather that have the incorrect return type.